website_sale_require_legal/controllers/main.py

```diff
@@ -27,16 +27,15 @@ def _mandatory_legal_terms(self):
     def checkout_form_validate(self, mode, all_form_values, data):
         """Require accepting legal terms to validate form."""
         # Patch context
-        old_context = request.context
-        request.context = dict(
-            request.context,
+        old_context = dict(request.context)
+        request.update_context(
             needs_legal=request.website.viewref(
                 "website_sale_require_legal.address_require_legal"
-            ).active,
+            ).active
         )
         result = super().checkout_form_validate(mode, all_form_values, data)
         # Unpatch context
-        request.context = old_context
+        request.update_context = old_context
```

Comment:
This is wrong. Update context is a method not a dict.

```suggestion
        request.update_env(context=old_context)
```

Reply:
Thanks, I was confused when I looked at another code, the thing is that while I was debugging it did update the context with the old one in that way. Anyway this would be the correct way.